apps/etl/transform/sources/desinventar.py

```diff
@@ -21,61 +18,32 @@ class DesinventarTransformHandler(BaseTransformerHandler[DesinventarTransformer,
     transformer_schema = DesinventarDataSource
 
     @classmethod
-    def get_schema_data(cls, extraction_obj: ExtractionData, country_code: str, iso3: str):  # type: ignore[reportIncompatibleMethodOverride]
+    def get_schema_data(cls, extraction_obj: ExtractionData):
+        from apps.etl.extraction.sources.desinventar.extract import DesInventarExtractionInputMetadata
+
+        metadata: dict | None = extraction_obj.metadata
+        if not metadata:
+            raise Exception("Metadata is not defined")
+        input_metadata = metadata.get("input", {})
+        input_metadata = DesInventarExtractionInputMetadata(
+            country_code=input_metadata.get("country_code"),
+            iso3=input_metadata.get("iso3"),
+        )
```

Comment:
```suggestion
        input_metadata = DesInventarExtractionInputMetadata.model_validate(input_metadata)
```

Reply:
Apply this method in all places where pydantic object is created from dictionnary.